src/main/java/io/vertx/redis/impl/AbstractRedisClient.java

```diff
@@ -190,23 +190,24 @@ final void sendJsonObject(final String command, final JsonArray args, final Hand
         break;
     }
 
-    Context context = vertx.getOrCreateContext();
-
-    Command<T> comm = new Command<T>(context, command, redisArgs, binary ? binaryCharset : charset)
+    Command<T> comm = new Command<T>(command, redisArgs, binary ? binaryCharset : charset)
                      .setExpectedReplies(expectedReplies).setHandler((Reply reply) -> {
         switch (reply.type()) {
           case '-': // Error
-            resultHandler.handle(new RedisAsyncResult<>(reply.asType(String.class)));
+            vertx.getOrCreateContext().runOnContext(v ->
```

Comment:
This doesn't look right to me. When this is invoked it will be on the context of the netsocket of the connection - that's probably going to be a different context to person who sent the original message.

I think it should invoked on the same context as he who sent the message.